service_mesh/v2x/ossm-extensions.adoc

```diff
@@ -0,0 +1,169 @@
+|===
+
+=== Example
+
+For a complete example that was built using the Rust SDK, take a look at the link:https://github.com/maistra/header-append-filter[header-append-filter]. It is a very simple filter that append a header, `custom-header`, to all responses, with the value depending on its configuration.
+
+=== Enabling WebAssembly Extension support
+
+Support for WebAssembly Extensions to {ProductName} is currently in Tech Preview, so it must be explicitly enabled for your `ServiceMeshControlPlane`. Set `spec.techPreview.wasmExtensions.enabled` in your SMCPv2 to `true`. Here's an example:
+
+[source,yaml]
+----
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: openid-connect
+  namespace: istio-system
+spec:
+  techPreview:
+    wasmExtensions:
+      enabled: true
+----
+
+=== Deploying Extensions
+
+{ProductName} Extensions can be enabled using the `ServiceMeshExtension` resource. The following snippet is an example resource.
+
+[source,yaml]
+----
+apiVersion: maistra.io/v1alpha1
+kind: ServiceMeshExtension
+metadata:
+  name: header-append
+  # Deploy to the Control Plane Namespace to enable globally, 
+  # or to another namespace to only enable locally
+  namespace: istio-system
+spec:
+  # Specify a set of labels to match specific workloads.
+  # Or, leave empty to match all workloads.
+  workloadSelector:
+    labels:
+      app: httpbin
+  # The configuration to be passed to the extension. 
+  # In this case, the value of the header that
+  # is added to the requests.
+  config: test
+  # The container image containing the extension
+  image: quay.io/maistra-dev/header-append-filter:latest
+  # Where in the filter chain this extension should be injected
+  # One of: PreAuthN, PostAuthN, PreAuthZ, PostAuthZ, PreStats, PostStats
+  phase: PostAuthZ.
+  # Priority is used for ordering of extensions within the same phase. 
+  # Higher priority means
+  # earlier execution.
+  priority: 100
+----
+
+.ServiceMeshExtension Field Reference
+|===
+| Field | Description
+
+|metadata.namespace
+|The `metadata.namespace` of a `ServiceMeshExtension` source has a special semantic: if it equals the Control Plane Namespace, the Extension will be applied to all workloads in the Service Mesh that match its `workloadSelector`. When deployed to any other Mesh Namespace, it will only be applied to workloads in that same Namespace.
+
+|spec.workloadSelector
+|The `spec.workloadSelector` field has the same semantic as the `spec.selector` field of the link:https://istio.io/v1.6/docs/reference/config/networking/gateway/#Gateway[Istio Gateway resource]. It will match a workload based on its Pod labels. If no `workloadSelector` is specified, the Extension will be applied to all workloads in the namespace.
+
+|spec.config
+|This is a pass-through string field that will be handed over to the Extension, so syntax and semantics are dependant on the Extension you're deploying.
+
+|spec.phase
```

Comment:
this one's a duplicate and can be removed

Reply:
I noticed the field is duplicated but the description is not. Just a heads up so you use the right description.